src/toil/fileStores/__init__.py

```diff
@@ -33,32 +34,34 @@ class FileID(str):
     def __new__(cls, fileStoreID, *args):
         return super(FileID, cls).__new__(cls, fileStoreID)
 
-    def __init__(self, fileStoreID, size):
+    def __init__(self, fileStoreID, size, executable=False):
         # Don't pass an argument to parent class's __init__.
         # In Python 3 we can have super(FileID, self) hand us object's __init__ which chokes on any arguments.
         super(FileID, self).__init__()
         self.size = size
+        self.executable = executable
 
     def pack(self):
         """
         Pack the FileID into a string so it can be passed through external code.
         """
-        return '{}:{}'.format(self.size, self)
+        return '{}:{}:{}'.format(self.size, int(self.executable), self)
 
     @classmethod
     def forPath(cls, fileStoreID, filePath):
-        return cls(fileStoreID, os.stat(filePath).st_size)
+        executable = os.stat(filePath).st_mode & stat.S_IXUSR != 0
+        return cls(fileStoreID, os.stat(filePath).st_size, executable)
 
     @classmethod
     def unpack(cls, packedFileStoreID):
         """
         Unpack the result of pack() into a FileID object.
         """
-
-        # Find the separating character (first instance)
-        sepPos = packedFileStoreID.find(':')
+        # Only separate twice in case the FileID itself has colons in it
+        vals = packedFileStoreID.split(':', 2)
```

Comment:
@adamnovak I think this should fix the issue with colons being in the `FileID` itself but it isn't one of the ways you recommended to go about it. Let me know if I'm not accounting for something.

Reply:
Yeah, that should work, I think.